app/src/main/java/io/neurolab/fragments/NeuroSettingsFragment.java

```diff
@@ -1,40 +1,56 @@
 package io.neurolab.fragments;
 
-import android.content.Context;
 import android.content.SharedPreferences;
 import android.os.Bundle;
-import com.takisoft.fix.support.v7.preference.PreferenceFragmentCompat;
 import android.support.annotation.Nullable;
 import android.support.v7.preference.EditTextPreference;
-import android.support.v7.preference.Preference;
-import android.support.v7.preference.PreferenceScreen;
-import android.widget.EditText;
+
+import com.takisoft.fix.support.v7.preference.PreferenceFragmentCompat;
 
 import io.neurolab.R;
 
-public class NeuroSettingsFragment extends PreferenceFragmentCompat implements SharedPreferences.OnSharedPreferenceChangeListener, Preference.OnPreferenceChangeListener {
-    Context context = getContext();
-    SharedPreferences sharedPreferences;
-    SharedPreferences.Editor editor;
+public class NeuroSettingsFragment extends PreferenceFragmentCompat implements SharedPreferences.OnSharedPreferenceChangeListener {
 
-    public NeuroSettingsFragment() {
+    public static final String KEY_SAMPLES = "samples";
+    public static final String KEY_BINS = "bins";
+    public static final String KEY_CHANNELS = "numChannels";
 
-    }
+    private EditTextPreference samplesPref;
+    private EditTextPreference binsPref;
+    private EditTextPreference channelsPref;
+
+    SharedPreferences sharedPreferences;
```

Comment:
we can make this private and use getters and setters for encapsulation

Reply:
For shared preferences, I don't think we need getters and setters. We can get any pref with required key and I believe there is a method like `getDefaultSharedPreferences` that can be used in any Activity. @suresh-gandhi What do you say? Though I agree we can make it private.

Reply:
@jddeep I agree with you and also meant the same.

Reply:
Ahh good catch. I reused the stuff there. Will update the PR..